pkg/rest/vf/vm_config.go

```diff
@@ -4,28 +4,30 @@ import (
 	"net/http"
 
 	"github.com/Code-Hex/vz/v3"
+	"github.com/crc-org/vfkit/pkg/config"
 	"github.com/crc-org/vfkit/pkg/rest/define"
 	"github.com/gin-gonic/gin"
 	"github.com/sirupsen/logrus"
 )
 
 type VzVirtualMachine struct {
-	VzVM   *vz.VirtualMachine
-	config *vz.VirtualMachineConfiguration
+	VzVM     *vz.VirtualMachine
+	config   *vz.VirtualMachineConfiguration
+	vmConfig *config.VirtualMachine
 }
 
-func NewVzVirtualMachine(vm *vz.VirtualMachine, config *vz.VirtualMachineConfiguration) *VzVirtualMachine {
-	return &VzVirtualMachine{config: config, VzVM: vm}
+func NewVzVirtualMachine(vm *vz.VirtualMachine, config *vz.VirtualMachineConfiguration, vmConfig *config.VirtualMachine) *VzVirtualMachine {
+	return &VzVirtualMachine{config: config, VzVM: vm, vmConfig: vmConfig}
 }
 
 // inspect returns information about the virtual machine like hw resources
 // and devices
 func (vm *VzVirtualMachine) Inspect(c *gin.Context) {
 	ii := define.InspectResponse{
-		// TODO complete me
-		CPUs:   1,
-		Memory: 2048,
-		//Devices: vm.Devices,
+		CPUs:   vm.vmConfig.Vcpus,
+		Memory: vm.vmConfig.MemoryBytes,
+		// TODO: add devices (VirtioDevice type needs to be improved) @Black-Hole1
```

Comment:
PR looks good to me without this added `TODO`

Reply:
Great :) was not sure if this was now waiting for feedback from me, or if it was still work in progress. Great to know you are going to improve it some more!